### ISSUE TITLE
Update Supported Fedora Repos

### ISSUE THREAD

**Issue by JKomoroski** [irontec/sngrep#341]
At this time [http://packages.irontec.com/fedora](http://packages.irontec.com/fedora) only lists fedora 20-25. Fedora 25 went end of life on December 12th, 2017. Can you add current Fedora release(s)?
I happen to be using the latest Fedora 33. I haven't tested your Fedora 25 repo on 33 yet.
An alternative may be to publish in a universal package format.

Thanks!

**Comment by Kaian** [comment]
Hi @JKomoroski 

Thanks for the input!

Actually, we have never used sngrep rpms, and our knowledge building them is very limited. We use [mock](https://packages.debian.org/buster/mock) for RPM building, and it seems to only support [configurations for some fedora versions](https://packages.debian.org/buster/all/mock/filelist).

Not sure if there is some other tool that could help with RPM building tasks, like pbuilder for debian packages.

**Comment by sergey-safarov** [comment]
Are you use docker?
I can prepare an instruction how to package using docker.

**Comment by Kaian** [comment]
The jenkins environment we use for packaging for debian has docker, although its not used for packaging.

I would love to see some sample Dockerfiles for building RPMs for sure!

**Comment by sergey-safarov** [comment]
Hi @Kaian 
I have prepared a script to package `sngrep` for different dists. Please check repo https://github.com/sergey-safarov/sngrep-builder

I provided `README.md` with instructions on how to use the builder image.
I ready to transfer the repo to you.

**Comment by Kaian** [comment]
Hi @sergey-safarov

Thanks a lot for your hard work, that's simply amazing!

I would love to move the templates and scripts into this project, maybe `pkg/docker` directory could be a good place for them.

Also, I've been testing the images and looks great, I will require some time to integrate them in Jenkins and add the repository part of each package (and maybe add some repos for alpine).

I have only encountered a problem so far with the red hat repositories. We don't have any RHEL subscritpion so after reading a bit about available repositories it looks like registry.access.redhat.com could work for retrieving base images without authentication.

https://access.redhat.com/RegistryAuthentication

Looks like base image can be pulled from that repo, but you still need credentials for using codeready repositories where development packages live. I hope there is some other way to download -devel packages from RHEL so anyone with this dockerfiles can setup its own automated builds without depending on any subscription.

Once again, thanks for your hard work!

**Comment by sergey-safarov** [comment]
About RedHat subscription.
I use a free developer license. This requires yearly renewing the subscription (also free).
https://developers.redhat.com/articles/getting-red-hat-developer-subscription-what-rhel-users-need-know

Once you get a subscription then you get access to required RPM repos.
During the build for RHEL also used RedHat docker registry for `ubi` images. To get access to these images need execute the command
```
docker login https://registry.redhat.io
```
https://access.redhat.com/RegistryAuthentication

**Comment by sergey-safarov** [comment]
I look at how to:
1. automate packaging using drone.io. This allows building not only for `x86` arch
2. push binary packages to the github releases page.

For me it not fast, but good options for the future.
Relevant links
1. https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/about-releases
2. https://docs.github.com/en/rest/reference/repos#releases

**Comment by msehnout** [comment]
I'd suggest to use https://copr.fedorainfracloud.org/ for building Fedora and CentOS packages. It is pretty easy and github releases can trigger the build.

**Comment by Kaian** [comment]
Hi!

I have created a project there and will try to keep it up to date with every release.
We rarely use the RPM packages for sngrep, so any help testing and documenting in the wiki would be appreciated.

https://copr.fedorainfracloud.org/coprs/irontec/sngrep/

Thanks!

**Comment by msehnout** [comment]
Hi! Perfect, I'll take a look.

**Comment by msehnout** [comment]
I've update the wiki with steps for Copr. I did not change RHEL and CentOS as those are still up-to-date.
https://github.com/irontec/sngrep/wiki/Installing-Binaries/_compare/fd23f637e0cf70e17b94abc83970be6d5a7a9227...b43e5e8e5d8ae84198d6a3f5855fc27020a7bd38

**Comment by Kaian** [comment]
I'm ok with removing irontec RPMs repos if Copr provide them and its distros are enough for the community.
As mentioned, we don't use those packages, so maybe we can just use Copr.

**Comment by msehnout** [comment]
Yeah, the copr repo should work for both CentOS and RHEL. It's just that RHEL isn't explicitly mentioned there so I decided to leave it in the wiki.
